Upstream submission:

  <https://sourceware.org/pipermail/libc-alpha/2021-June/128163.html>

Author: Florian Weimer <fweimer@redhat.com>
Date:   Mon Jun 28 10:05:44 2021 +0200

    nptl_db: Clean up main/rtld variable handling
    
    Most symbols are now in libc.so.6.  The "main" (exempted from
    coverage checks) status is therefore not necessary.  Use
    DB_MAIN_VARIABLE for the remaining separate symbol,
    __nptl_initial_report_events.  DB_RTLD_VARIABLE is now unused, so
    remove it.

diff --git a/nptl_db/db-symbols.awk b/nptl_db/db-symbols.awk
index 6f326cf379fc886b..ef1d91b167998d3d 100644
--- a/nptl_db/db-symbols.awk
+++ b/nptl_db/db-symbols.awk
@@ -2,7 +2,6 @@
 # we've just built.  It checks for all the symbols used in td_symbol_list.
 
 BEGIN {
-%define DB_RTLD_VARIABLE(name) /* Nothing. */
 %define DB_MAIN_VARIABLE(name) /* Nothing. */
 %define DB_MAIN_SYMBOL(name) /* Nothing. */
 %define DB_MAIN_ARRAY_VARIABLE(name) /* Nothing. */
diff --git a/nptl_db/structs.def b/nptl_db/structs.def
index 6a726f207eca49a3..fb7bb9367d7c01ec 100644
--- a/nptl_db/structs.def
+++ b/nptl_db/structs.def
@@ -22,10 +22,6 @@
 # define STRUCTS_DEF_DEFAULTS 1
 #endif
 
-#ifndef DB_RTLD_VARIABLE
-# define DB_RTLD_VARIABLE(name) DB_VARIABLE (name)
-#endif
-
 /* DB_MAIN_VARIABLE, DB_MAIN_SYMBOL, DB_MAIN_ARRAY_VARIABLE are not
    covered by the libc symbol check in db-symbols.awk.  */
 #ifndef DB_MAIN_VARIABLE
@@ -78,14 +74,14 @@ DB_STRUCT_FIELD (td_eventbuf_t, eventnum)
 DB_STRUCT_FIELD (td_eventbuf_t, eventdata)
 
 DB_SYMBOL (nptl_version)
-DB_MAIN_SYMBOL (__nptl_create_event)
-DB_MAIN_SYMBOL (__nptl_death_event)
+DB_SYMBOL (__nptl_create_event)
+DB_SYMBOL (__nptl_death_event)
 DB_SYMBOL (__nptl_threads_events)
-DB_MAIN_VARIABLE (__nptl_nthreads)
+DB_VARIABLE (__nptl_nthreads)
 DB_VARIABLE (__nptl_last_event)
-DB_RTLD_VARIABLE (__nptl_initial_report_events)
+DB_MAIN_VARIABLE (__nptl_initial_report_events)
 
-DB_MAIN_ARRAY_VARIABLE (__pthread_keys)
+DB_ARRAY_VARIABLE (__pthread_keys)
 DB_STRUCT (pthread_key_struct)
 DB_STRUCT_FIELD (pthread_key_struct, seq)
 DB_STRUCT_FIELD (pthread_key_struct, destr)
